### PR TITLE
Add Global Top 10 Leaderboard

### LINKED ISSUE #107: Create global leaderboard (Top 10)
- [ ] Make all new users auto-join that leaderboard
- [ ] Limit that leaderboard to the top 10 users

### PR DESCRIPTION
**Pull request by trouni**
### Modifications to the Leaderboard model

Four attributes have been added to the Leaderboard model:
- `description` (could be used by the FE)
- `auto_join` (whether or not to join the leaderboard upon signup)
- `leave_disabled` (disable the leave option)
- `rankings_top_n` (limit the number of users returned for the leaderboard to the top n)

### Refactoring of seeds

The seeds have been refactored in order to allow using multiple seeds files inside the `db/seeds` folder. `rake db:seed` will run all the files inside the `db/seeds` folder. Since the order of execution matters, I have moved the content of the previous `seeds.rb` file into a `0_default.rb` file, but we could also split this into separate files in the future.

In order to seed the global leaderboard, we should run this rake task:
```sh
rake db:seed:1_leaderboards
```

## Remaining tasks

Some changes still need to be made in the FE:
- [x] Hide the "Leave leaderboard" button if the `leave_disabled` flag is `true` (Fixed in trouni/predictor-vue#234)

## Related issues

Fixes #107, fixes trouni/predictor-vue#215

### CODE REVIEW

**Comment by trouni** [comment]
@dmbf29 Don't pay attention to the long commit history, I didn't branch out from the most recent main... Only the last 2 commits are relevant to this PR.

**Comment by dmbf29** [comment]
takes 3 minutes to create memberships for every competition 🤔 should take seconds...

File: app/controllers/auth/devise_token_auth/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Auth
+  module DeviseTokenAuth
+    class RegistrationsController < ::DeviseTokenAuth::RegistrationsController
+      after_action :auto_join_leaderboards, only: :create
```
[the remark this answers is not in the document]
Comment (trouni) [reply app/controllers/auth/devise_token_auth/registrations_controller.rb:4]: Yup you're right! I'll stick to an `after_create` callback!

**Comment by dmbf29** [comment]
looks like the leaderboards are working but the names don't change anymore when i swipe left to right

Screen Shot 2022-12-07 at 01 15 25

**Review by dmbf29 — APPROVED** [review]
LGTM 👍

**Pull request merged**